sympy/polys/rootoftools.py

```diff
@@ -1004,6 +1004,25 @@ def eval_rational(self, dx=None, dy=None, n=15):
         self._set_interval(interval)
         return real + I*imag
 
+    def as_real_imag(self, deep=True, **hints):
+        """Return the real and imaginary part of the root."""
+        if self.is_real:
+            return self, S.Zero
+        if self.is_imaginary:
+            return S.Zero, self
+        x = Dummy('x')
+        y = Dummy('y')
+        p = self.poly.as_poly()
+        p_re, p_im = p.real_imag(x, y)
+        p_x = resultant(p_re, p_im, y).as_poly(x)
+        p_y = resultant(p_re, p_im, x).as_poly(y)
+        for rx in p_x.real_roots():
+            for ry in p_y.real_roots():
+                if p.same_root(self, rx + I*ry):
+                    return rx, ry
```

Comment:
This could more efficiently check the real and imaginary parts separately like:
```
for rx in p_x.real_roots():
    if p_x.same_root(rx, re(self)):
```